diff --git a/Rscripts/form_transcripts.R b/Rscripts/form_transcripts.R
index 292a505..7631df8 100755
--- a/Rscripts/form_transcripts.R
+++ b/Rscripts/form_transcripts.R
@@ -37,7 +37,8 @@ arguments <- parse_args(OptionParser(usage = "%prog [options] counts_file groups
    make_option(c("-o","--out_prefix"), default = sprintf("%s",getwd()), help="The output directory for the kmer data"),
    make_option(c("-t","--txdb"), default=NULL, help="The txdb object"),
    make_option(c("-j","--juncs"), default=NULL, help="The junction file (path and file)"),
-   make_option(c("-b","--bsgenome_name"),default = "bsgenome_name",help="the bsgenome object name"),
+   make_option(c("-b","--bsgenome_name"),default = "Hsapiens",help="the BSgenome object name"),
+   make_option(c("-p","--bsgenome_package"),default = "BSgenome.Hsapiens.UCSC.hg38",help="the BSgenome package name"),
    make_option(c("-m","--chr_map"),default=F,help="the chromosome map"),
    make_option(c("-f","--funcs"),default="",help="functions"))))
 
@@ -53,8 +54,9 @@ print(opt$funcs)
 source(opt$funcs)
 
 print("loading Bsgenome")
-bsgenome_name <- opt$bsgenome_name # bsgenome_name so that can create bsgenome object
-library(bsgenome_name,character.only = T) # assigning bsgenome object to "bsgenome" variable
-assign("bsgenome",get(bsgenome_name))
+bsgenome_name <- opt$bsgenome_name
+bsgenome_package <- opt$bsgenome_package
+library(bsgenome_package,character.only = T)
+assign("bsgenome", getExportedValue(bsgenome_package, bsgenome_name))
 
 print("reading in splice-junctions")
